pkg/controller/logstorage/users/user_controller.go

```diff
@@ -0,0 +1,254 @@
+		pw, err := randomPassword(16)
+		if err != nil {
+			r.status.SetDegraded(operatorv1.ResourceReadError, "Error generating Linseed password", err, reqLogger)
+			return reconcile.Result{}, err
+		}
+		basicCreds = corev1.Secret{}
+		basicCreds.Name = render.ElasticsearchLinseedUserSecret
+		basicCreds.Namespace = helper.TruthNamespace()
+		basicCreds.StringData = map[string]string{"username": linseedUser.Username, "password": pw}
+
+		// Make sure we install the generated credentials into the truth namespace.
+		credentialSecrets = append(credentialSecrets, &basicCreds)
+	}
+	if helper.TruthNamespace() != helper.InstallNamespace() {
+		// Copy the credentials into the install namespace.
+		credentialSecrets = append(credentialSecrets, secret.CopyToNamespace(helper.InstallNamespace(), &basicCreds)[0])
+	}
+	credentialComponent := render.NewPassthrough(credentialSecrets...)
+
+	// In standard installs, the LogStorage owns the secret. For multi-tenant, it's owned by the tenant.
+	var hdler utils.ComponentHandler
+	if r.multiTenant {
+		hdler = utils.NewComponentHandler(reqLogger, r.client, r.scheme, tenant)
+	} else {
+		hdler = utils.NewComponentHandler(reqLogger, r.client, r.scheme, logStorage)
+	}
+	if err := hdler.CreateOrUpdateOrDelete(ctx, credentialComponent, r.status); err != nil {
+		r.status.SetDegraded(operatorv1.ResourceUpdateError, "Error creating / updating Linseed user secret", err, reqLogger)
+		return reconcile.Result{}, err
+	}
+
+	// Now that the secret has been created, also provision the user in ES.
+	if err := r.createLinseedLogin(ctx, tenantID, &basicCreds, reqLogger); err != nil {
+		r.status.SetDegraded(operatorv1.ResourceUpdateError, "Failed to create Linseed user in ES", err, reqLogger)
+		return reconcile.Result{}, err
+	}
+
+	r.status.ReadyToMonitor()
+	r.status.ClearDegraded()
+	return reconcile.Result{}, nil
+}
+
+func (r *UserController) createLinseedLogin(ctx context.Context, tenantID string, secret *corev1.Secret, reqLogger logr.Logger) error {
+	// ES should be in ready phase when execution reaches here, apply ILM polices
+	esClient, err := utils.NewElasticClient(r.client, ctx, relasticsearch.ElasticEndpoint())
+	if err != nil {
+		r.status.SetDegraded(operatorv1.ResourceCreateError, "Failed to connect to Elasticsearch - failed to create the Elasticsearch client", err, reqLogger)
+		return err
+	}
+
+	// Determine the password from the secret.
+	password := secret.StringData["password"]
+	if password == "" {
+		password = string(secret.Data["password"])
+	}
+	if password == "" {
+		return fmt.Errorf("Unable to find password in secret")
+	}
+
+	// Create the user in ES.
+	user := utils.LinseedUser(tenantID)
+	user.Password = password
+	if err = esClient.CreateUser(ctx, user); err != nil {
+		r.status.SetDegraded(operatorv1.ResourceUpdateError, "Failed to create or update Elasticsearch user", err, reqLogger)
+		return err
+	}
+
+	return nil
+}
+
+func randomPassword(length int) (string, error) {
```

Comment:
We have a function in the code already for generating a password: GeneratePassword() I think it is best to have only one.

! Passwords should be created using crypto/rand